Documentation/articles/command/command_handler.md

```diff
@@ -0,0 +1,44 @@
+---
+title: Handling Commands
+description: Recommendations and Requirements when Handling a Command
+keywords: 
+author: einari, smithmx
+---
+
+# Command Handler
+
+## Background
+
+A *Command Handler* handles commands.  The command handler is responsible for executing the appropriate methods on Aggregate Roots and Domain Services to fulfill the intent of the *Command*.
+
+## Requirements
+
+A *Command Handler* must implement the *IHandleCommands* interface.  This is a simple marker interface with no methods to implement.
+
+To handle a *Command*, you create a method called *Handle* with a single parameter that is the *Command* you wish to handle.  A *Command Handler* does not return anything.  It is not permitted to return a value from a *Command*.  When a *Command* is executed, you will receive a *CommandResult*
+which will indicate whether the *Command* succeeded and, in the case of failure, why the *Command* failed.  No artifacts that were created or updated as a result of the *Command* will be present on the *CommandResult*.
+
+```csharp
+public void Handle(AddRecommendationToCart cmd)
+```
+
+There **MUST** be a single handler for a Command.  Dolittle will throw an exception if there is no handler for a command or if there is more than one handler for a command.  There can only be a single handler for a command as there must be a clear and unambiguous indication of the result of the execution of the command.  Multiple command handlers for a single command would require co-ordination of multiple results with associated rollback of successful handlers after unsuccessful ones.  Where you wish multiple things to happen on receipt of a Command, you must implement and handle this yourself in your *Command Handler*.
+
+## Responsibilities
+
+The sole responsibility of the *Command Handler* is to execute the *Command*.
+
+A *Command Handler* should not implement any validation, security checks, or such like.  These should be handled earlier in the command pipeline by validators.  When we invoke a *Command Handler*, it is with the expectation that the action will succeed.  We should have validated the command,
+checked business rule conditions, checked the permissions and therefore the command should succeed.  It is for this reason that within a *Command Handler*, the only way to indicate that the *Command* did not succeed is via an Exception.  This is because, due to our previous checks,
+it is truly an exceptional condition for the *Command* to fail.
+
+## Design
+
+It is recommended that a *Command Handler* should invoke a single method on an *Aggregate Root*.  This is an indication that you have correctly modelled your *Aggregate Root*, *Command* and business transaction.  It is sometimes useful to think of a *Command* as a serialized call on a method on an *Aggregate Root*. If you are calling multiple *Aggregate Roots* you will need to handle one of them failing and rolling back the transaction.
```

Comment:
Refer to the comment above.  This will actually not work going forward.  Bifrost supported tracking multiple aggregates but Dolittle won't... because of the Event Store / Commit metaphor.